src/main/java/com/couchbase/lite/store/SQLiteViewStore.java

```diff
@@ -406,61 +505,79 @@ public void emit(Object key, Object value) {
                     }
                 }
 
-                if (deleted) {
+                if (deleted)
                     continue;
-                }
 
+                // Get json blob:
                 String[] selectArgs3 = {Long.toString(sequence)};
                 byte[] json = SQLiteUtils.byteArrayResultForQuery(store.getStorageEngine(),
                         "SELECT json FROM revs WHERE sequence=?", selectArgs3);
 
                 // Get the document properties, to pass to the map function:
-                Map<String, Object> properties = store.documentPropertiesFromJSON(
+                Map<String, Object> curDoc = store.documentPropertiesFromJSON(
                         json,
                         docId,
-                        revId,
+                        revID,
                         false,
                         sequence
                 );
-                if (properties != null) {
-                    //TODO checkDocTypes here
 
-                    // Call the user-defined map() to emit new key/value
-                    // pairs from this revision:
-                    emitBlock.setSequence(sequence);
-                    delegate.getMap().map(properties, emitBlock);
-
-                    properties.clear();
+                if (curDoc == null) {
+                    Log.w(Log.TAG_VIEW, "Failed to parse JSON of doc %s rev %s", docID, revID);
+                    continue;
+                }
+                curDoc.put("_local_seq", sequence);
+
+                if (conflicts != null)
+                    curDoc.put("_conflicts", conflicts);
+
+                // Call the user-defined map() to emit new key/value pairs from this revision:
+                i = -1;
+                for (SQLiteViewStore view : views) {
+                    ++i;
+                    if (viewLastSequence[i] < realSequence) {
+                        if (checkDocTypes) {
+                            String viewDocType = viewDocTypes.get(view.getName());
+                            if (viewDocType != null && !viewDocType.equals(docType))
+                                continue; // skip; view's documentType doesn't match this doc
+                        }
+                    }
+                    Log.v(Log.TAG_VIEW, "#%s: map '%s' for view %s...",
+                            sequence, docID, view.getName());
+                    try {
+                        emitBlock.setSequence(sequence);
+                        delegate.getMap().map(curDoc, emitBlock);
```

Comment:
I think `map` function is per View. Retrieving `map` function from `delegate` might apply different `map` function??